tools/distpartitioning/dataset_utils.py

```diff
@@ -33,46 +33,63 @@ def get_dataset(input_dir, graph_name, rank, num_node_weights):
     """
     #node features dictionary
     node_features = {}
+    node_feature_tids = {}
 
-    #iterate over the sub-dirs and extract the nodetypes
-    #in each nodetype folder read all the features assigned to 
-    #current rank
-    siblings = os.listdir(input_dir)
-    for s in siblings:
-        if s.startswith("nodes-"):
-            tokens = s.split("-")
-            ntype = tokens[1]
-            num_feats = tokens[2]
-            for idx in range(int(num_feats)):
-                feat_file = s +'/node-feat-'+'{:02d}'.format(idx) +'/'+ str(rank)+'.npy'
-                if (os.path.exists(input_dir+'/'+feat_file)):
-                    features = np.load(input_dir+'/'+feat_file)
-                    node_features[ntype+'/feat'] = torch.tensor(features)
+    #iterate over the "node_data" dictionary in the schema_map
+    #read the node features if exists
+    #also keep track of the type_nids for which the node_features are read.
+    dataset_features = schema_map["node_data"]
+    for ntype_name, ntype_feature_data in dataset_features.items():
+        #ntype_feature_data is a dictionary
+        #where key: feature_name, value: list of lists
+        node_feature_tids[ntype_name] = []
+        for feat_name, feat_data in ntype_feature_data.items():
+            assert len(feat_data) == world_size
+            my_feat_data = feat_data[rank]
+            if (os.path.isabs(my_feat_data[0])):
+                node_features[ntype_name+'/'+feat_name] = torch.from_numpy(np.load(my_feat_data[0]))
+            else:
+                node_features[ntype_name+'/'+feat_name] = torch.from_numpy(np.load(input_dir+my_feat_data[0]))
 
-    #done build node_features locally. 
-    if len(node_features) <= 0: 
-        print('[Rank: ', rank, '] This dataset does not have any node features')
-    else: 
-        for k, v in node_features.items():
-            print('[Rank: ', rank, '] node feature name: ', k, ', feature data shape: ', v.size())
+            v = [feat_name, my_feat_data[1], my_feat_data[2]]
+            l = node_feature_tids[ntype_name]
+            l.append(v)
+            node_feature_tids[ntype_name] = l
 
-    #read (split) xxx_nodes.txt file
-    node_file = input_dir+'/'+graph_name+'_nodes'+'_{:02d}.txt'.format(rank)
-    node_data = np.loadtxt(node_file, delimiter=' ', dtype='int64')
-    nodes_datadict = {}
-    nodes_datadict[constants.NTYPE_ID] = node_data[:,0]
-    type_idx = 0 + num_node_weights + 1
-    nodes_datadict[constants.GLOBAL_TYPE_NID] = node_data[:,type_idx]
-    print('[Rank: ', rank, '] Done reading node_data: ', len(nodes_datadict), nodes_datadict[constants.NTYPE_ID].shape)
+    #read my nodes for each node type
+    node_tids = {}
+    node_data = schema_map["nid"]
+    for ntype_name, ntype_info in node_data.items():
+        v = []
+        node_file_info = ntype_info["data"]
+        for idx in range(len(node_file_info)):
+            v.append((node_file_info[idx][1], node_file_info[idx][2]))
```

Comment:
here it still suggests users need to define node metadata files.

Reply:
Yes. As I mentioned already removing "nid" information from the metadata is not that straight forward. To completely remove it we must first define a way to compute the global node-ids for the node features which exist in the dataset. If the dataset does not have node-features in it then "nid" key-value pair can be completely removed from the metadata json file.